tests/test_clipboard.py

```diff
@@ -16,55 +16,42 @@ def test_unicode_clipboard():
 example_html = textwrap.dedent(
     """
     Version:0.9
-    StartHTML:71
-    EndHTML:170
-    StartFragment:140
-    EndFragment:160
-    StartSelection:140
-    EndSelection:160
-    <!DOCTYPE>
-    <HTML>
-    <HEAD>
-    <TITLE> The HTML Clipboard</TITLE>
-    <BASE HREF="http://sample/specs">
-    </HEAD>
-    <BODY>
-    <UL>
-    <!--StartFragment -->
-    <LI> The Fragment </LI>
-    <!--EndFragment -->
-    </UL>
-    </BODY>
-    </HTML>
+    StartHTML:00000138
+    EndHTML:00000215
+    StartFragment:00000171
+    EndFragment:00000182
+    StartSelection:00000174
+    EndSelection:00000178
+    <html><body>
+    <!--StartFragment--><p>😀</p><!--EndFragment-->
+    </body></html>
     """
 ).lstrip()
-"""
-Example taken from
-https://msdn.microsoft.com/en-us/library/windows/desktop/ms649015(v=vs.85).aspx#Description
-"""
 
 
 @pytest.fixture
 def sample_html():
     with wc.context():
+        wc.EmptyClipboard()
         wc.SetClipboardData(api.CF_HTML, example_html.encode('utf-8'))
 
 
 def test_html_paste(sample_html):
     res = wc.get_html()
-    assert len(res.html) < len(res.data)
-    assert res.headers == dict(
-        StartHTML=71,
-        Version=0.9,
-        EndHTML=170,
-        StartFragment=140,
-        EndFragment=160,
-        StartSelection=140,
-        EndSelection=160,
+    assert (
+        res.html
+        == '<html><body>\n<!--StartFragment--><p>😀</p><!--EndFragment-->\n</body></html>'
     )
+    assert res.fragment == '<p>😀</p>'
+    assert res.selection == '😀'
 
 
-@pytest.mark.xfail(reason="example is buggy")
-def test_html_fragment(sample_html):
-    snippet = wc.get_html()
-    assert snippet.fragment == '<LI> The Fragment </LI>'
```

Comment:
Probably we should keep this test... at least until we understand why it's failing. I can't remember why I created it, but it does look like some expectation is being missed that shouldn't be.

Reply:
It seems that offsets in `example_html` are corrupted.